src/slider-new/helpers/useMove.ts

```diff
@@ -0,0 +1,246 @@
+}
+
+export function useMove(props: MoveProps): HTMLAttributes<HTMLElement> {
+  const { onMoveStart, onMove, onMoveEnd } = props;
+
+  const state = useRef<{
+    didMove: boolean;
+    lastPosition: { pageX: number; pageY: number } | null;
+    id: number | null;
+  }>({ didMove: false, lastPosition: null, id: null });
+
+  const { addGlobalListener, removeGlobalListener } = useGlobalListeners();
+
+  const moveProps = useMemo(() => {
+    const moveProps: HTMLAttributes<HTMLElement> = {};
+
+    const start = () => {
+      disableTextSelection();
+      state.current.didMove = false;
+    };
+    const move = (
+      pointerType: BaseMoveEvent["pointerType"],
+      deltaX: number,
+      deltaY: number,
+    ) => {
+      if (!state.current.didMove) {
+        state.current.didMove = true;
+        onMoveStart?.({
+          type: "movestart",
+          pointerType,
+        });
+      }
+      onMove({
+        type: "move",
+        pointerType,
+        deltaX: deltaX,
+        deltaY: deltaY,
+      });
+    };
+    const end = (pointerType: BaseMoveEvent["pointerType"]) => {
+      restoreTextSelection();
+      if (state.current.didMove) {
+        onMoveEnd?.({
+          type: "moveend",
+          pointerType,
+        });
+      }
+    };
+
+    if (typeof PointerEvent === "undefined") {
+      const onMouseMove = (e: MouseEvent) => {
+        if (e.button === 0) {
+          if (isNull(state.current.lastPosition)) return;
+          move(
+            "mouse",
+            e.pageX - state.current.lastPosition.pageX,
+            e.pageY - state.current.lastPosition.pageY,
+          );
+          state.current.lastPosition = { pageX: e.pageX, pageY: e.pageY };
+        }
+      };
+      const onMouseUp = (e: MouseEvent) => {
+        if (e.button === 0) {
+          end("mouse");
+          removeGlobalListener(window, "mousemove", onMouseMove, false);
+          removeGlobalListener(window, "mouseup", onMouseUp, false);
+        }
+      };
+      moveProps.onMouseDown = (e: React.MouseEvent) => {
+        if (e.button === 0) {
+          start();
+          e.stopPropagation();
+          e.preventDefault();
+          state.current.lastPosition = { pageX: e.pageX, pageY: e.pageY };
+          addGlobalListener(window, "mousemove", onMouseMove, false);
+          addGlobalListener(window, "mouseup", onMouseUp, false);
+        }
+      };
+
+      const onTouchMove = (e: TouchEvent) => {
+        // @ts-ignore
+        const touch = [...e.changedTouches].findIndex(
+          ({ identifier }) => identifier === state.current.id,
+        );
+        if (touch >= 0) {
+          const { pageX, pageY } = e.changedTouches[touch];
+          if (isNull(state.current.lastPosition)) return;
+          move(
+            "touch",
+            pageX - state.current.lastPosition.pageX,
+            pageY - state.current.lastPosition.pageY,
+          );
+          state.current.lastPosition = { pageX, pageY };
+        }
+      };
```

Comment:
I have no idea if this code would work, i'm not able to correctly test it since the touch behaviour is broken in my side but we can try it later.

```ts
      const onTouchMove = (e: TouchEvent) => {
        // const touch = Array.from(e.changedTouches).findIndex(
        //   ({ identifier }) => identifier === state.current.id,
        // );
        const touch = e.changedTouches.item(state.current.id as number);
        if (touch) {
          const { pageX, pageY } = touch;
          if (isNull(state.current.lastPosition)) return;
          move(
            "touch",
            pageX - state.current.lastPosition.pageX,
            pageY - state.current.lastPosition.pageY,
          );
          state.current.lastPosition = { pageX, pageY };
        }
      };
```